core/src/main/scala/better/files/ThreadBackedFileMonitor.scala

```diff
@@ -46,7 +48,10 @@ abstract class ThreadBackedFileMonitor(val root: File, maxDepth: Int) extends Fi
     } else {
       when(file.exists)(file.parent).iterator  // There is no way to watch a regular file; so watch its parent instead
     }
-    toWatch.foreach(_.register(service))
+
+    toWatch.foreach { f =>
+      try { f.register(service) } catch { case NonFatal(e) => onException(e) }
```

Comment:
The `Try` is not used intentionally, because it would generate many many `Try` instances (one for each file) which is not very effective.